src/setup/node.rs

```diff
@@ -98,12 +98,13 @@ impl NodeBuilder {
 
         match node_type {
             NodeType::Stateful => {
-                let node_idx = if self.stateful_nodes_pool != 0 {
-                    self.stateful_nodes_pool -= 1;
-                    self.stateful_nodes_pool
+                let node_idx = if self.stateful_nodes_counter < STATEFUL_NODES_COUNT {
```

Comment:
nit; it would be shorter if we did `assert!(self.stateful_nodes_counter <= STATEFUL_NODES_COUNT);` after `self.stateful_nodes_counter += 1;`

Reply:
Done